File: CS/CustomizeNavigationControl.Module.Web/Controllers/WebCustomizeNavBarController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.Web.ASPxNavBar;
using DevExpress.ExpressApp.Web.Templates.ActionContainers;
using DevExpress.Web.ASPxTreeView;

namespace CustomizeNavigationControl.Module.Web.Controllers {
    public class WebCustomizeNavBarController : WindowController {
        public WebCustomizeNavBarController() {
            TargetWindowType = WindowType.Main;
        }
        protected override void OnActivated() {
            base.OnActivated();
            Frame.ProcessActionContainer += Frame_ProcessActionContainer;
        }
        private void Frame_ProcessActionContainer(object sender, ProcessActionContainerEventArgs e) {
            if(e.ActionContainer is NavigationActionContainer) {
                ASPxNavBar navBar = ((NavigationActionContainer)e.ActionContainer).NavigationControl as ASPxNavBar;
                if(navBar != null) {
                    // Customize the main ASPxNavBar control.
                    navBar.EnableAnimation = true;
                    foreach(NavBarGroup group in navBar.Groups) {
                        foreach(NavBarItem item in group.Items) {
                            if(item is NavBarTreeViewItem) {
                                ASPxTreeView innerTreeView = ((NavBarTreeViewItem)item).TreeViewNavigationControl.Control;
                                // Customize the inner ASPxTreeView control inside the navigation group.
                                innerTreeView.ShowExpandButtons = false;
                            }
                        }
                    }
                }
                else {
                    ASPxTreeView mainTreeView = ((NavigationActionContainer)e.ActionContainer).NavigationControl as ASPxTreeView;
                    if(mainTreeView != null) {
                        // Customize the main ASPxTreeView control.
                        mainTreeView.ShowExpandButtons = false;
                    }
                }
            }
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            Frame.ProcessActionContainer -= Frame_ProcessActionContainer;
        }
    }
}

File: VB/CustomizeNavigationControl.Module.Web/Controllers/WebCustomizeNavBarController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.Web
Imports DevExpress.ExpressApp.Web.Templates.ActionContainers
Imports DevExpress.ExpressApp.SystemModule

Namespace CustomizeNavigationControl.Module.Web.Controllers
    Public Class WebCustomizeNavBarController
        Inherits WindowController

        Public Sub New()
            TargetWindowType = WindowType.Main
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            AddHandler Frame.GetController(Of ShowNavigationItemController)().ShowNavigationItemAction.CustomizeControl, AddressOf ShowNavigationItemAction_CustomizeControl
        End Sub
        Private Sub ShowNavigationItemAction_CustomizeControl(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.Actions.CustomizeControlEventArgs)
            Dim navBar As ASPxNavBar = TryCast(e.Control, ASPxNavBar)
            If navBar IsNot Nothing Then
                ' Customize the main ASPxNavBar control.
                navBar.EnableAnimation = True
                For Each group As NavBarGroup In navBar.Groups
                    For Each item As NavBarItem In group.Items
                        If TypeOf item Is NavBarTreeViewItem Then
                            Dim innerTreeView As ASPxTreeView = CType(item, NavBarTreeViewItem).TreeViewNavigationControl.Control
                            ' Customize the inner ASPxTreeView control inside the navigation group.
                            innerTreeView.ShowExpandButtons = False
                        End If
                    Next item
                Next group
            Else
                Dim mainTreeView As ASPxTreeView = TryCast(e.Control, ASPxTreeView)
                If mainTreeView IsNot Nothing Then
                    ' Customize the main ASPxTreeView control.
                    mainTreeView.ShowExpandButtons = False
                End If
            End If
        End Sub
        Protected Overrides Sub OnDeactivated()
            MyBase.OnDeactivated()
            RemoveHandler Frame.GetController(Of ShowNavigationItemController)().ShowNavigationItemAction.CustomizeControl, AddressOf ShowNavigationItemAction_CustomizeControl
        End Sub
    End Class
End Namespace
